src/main/scala/wandou/avpath/Evaluator.scala

```diff
@@ -16,70 +16,97 @@ import wandou.avpath.Parser.Syntax
 import wandou.avpath.Parser.UnaryExprSyntax
 import wandou.avpath.Parser.PosSyntax
 import java.nio.ByteBuffer
+import java.util.regex.Pattern
+
 import org.apache.avro.Schema
 import org.apache.avro.Schema.Type
 import org.apache.avro.generic.GenericEnumSymbol
 import org.apache.avro.generic.GenericFixed
 import org.apache.avro.generic.IndexedRecord
+
 import scala.collection.mutable
 
 object Evaluator {
+
   sealed trait Target
+
   final case class TargetRecord(record: IndexedRecord, field: Schema.Field) extends Target
+
   final case class TargetArray(array: java.util.Collection[_], idx: Int, arraySchema: Schema) extends Target
+
   final case class TargetMap(map: java.util.Map[String, _], key: String, mapSchema: Schema) extends Target
 
   private object Op {
+
     case object Select extends Op
+
     case object Update extends Op
+
     case object Delete extends Op
+
     case object Clear extends Op
+
     case object Insert extends Op
+
     case object InsertAll extends Op
+
   }
+
   private sealed trait Op
 
-  final case class Ctx(value: Any, schema: Schema, topLevelField: Schema.Field, target: Option[Target] = None)
+  val pattern: Pattern = Pattern.compile("\\[(\\*|:-?\\d+|-?\\d+:|-?\\d+:-?\\d+)\\]")
```

Comment:
val date = """(\d\d\d\d)-(\d\d)-(\d\d)""".r
```suggestion
  import scala.util.matching.Regex
  // Matches any [*], [:n], or [n:m] range selectors.
  val pattern: Regex = """\[(\*|:-?\d+|-?\d+:|-?\d+:-?\d+)\\]""".re
```

It looks like the scala regexs are pretty clean for use in scala.